RentACar/RentACar/Repositories/ClientCRUD.cs

```diff
@@ -0,0 +1,54 @@
+﻿using RentACar.Models;
+using RentACar.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace RentACar.Repositories
+{
+    public class ClientCRUD : Repository<Client>, IClientCRUD
+    {
+
+        private readonly string _jsonFile;
+
+        public ClientCRUD(FilePathsStorage config) : base(config.Client)
+        {
+            _jsonFile = config.Client;
+        }
+
+        public override Client Create(Client client)
+        {
+            var clients = base.GetAll();
+
+            if (clients.Any(e => e.Dni == client.Dni))
+                return null;
+
+            client.LastUpdate = DateTime.UtcNow;
+
+            var clientCreated = base.Create(client);
+
+            return clientCreated;
+        }
+
+        public override List<Client> GetAll()
+        {
+            var list = base.GetAll()
+                           .OrderBy(e => e.Id)
+                           .ToList();
+            return list;
+        }
+
+        protected override void UpdateObject(Client existingObject, Client newObject)
+        {
+            existingObject.Dni = newObject.Dni;
```

Comment:
Im not sure if i have to update Dni, i'm thinking in the case that the Dni is updated by one that already exist.

Reply:
maybe you can check if there's already one with that value and do something about it.. beware WHERE you put such validation ;)

Reply:
oh, im scared now.. Hahha.. kidding.. thanks! i will make the validation and i notify you for feedback !